routing/router.go

```diff
@@ -1753,6 +1753,40 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 				"htlc=%x", errSource.SerializeCompressed(),
 				payment.PaymentHash[:])
 
+			// Always determine chan id ourselves, because a channel
+			// update with id may not be available.
+			failedChanID, err := getFailedChannelID(route, errSource)
+			if err != nil {
+				return preImage, nil, err
+			}
+
+			// processChannelUpdateAndRetry is a closure that
+			// handles a failure message containing a channel
+			// update. This function always tries to apply the
+			// channel update and passes on the result to the
+			// payment session to adjust its view on the reliability
+			// of the network.
+			//
+			// As channel id, the locally determined channel id is
+			// used. It does not rely on the channel id that is part
+			// of the channel update message, because the remote
+			// node may lie to us or the update may be corrupt.
+			processChannelUpdateAndRetry := func(
+				update *lnwire.ChannelUpdate,
+				pubKey *btcec.PublicKey) {
+
+				// Try to apply the channel update. If
+				// it fails, prune the edge. There is no
+				// reason to continue trying this channel.
+				err := r.applyChannelUpdate(update, pubKey)
```

Comment:
Now that this is the only place where the error is checked, could we return the `updateOk` bool directly?

Reply:
Good suggestion, fixed.